data/scenarios/Challenges/capture.yaml

```diff
@@ -0,0 +1,185 @@
+version: 1
+name: Pig wrestling
+author: Karl Ostmo
+description: |
+  Capture a slippery pig
+creative: false
+seed: 0
+attrs:
+  - name: bacon
+    fg: "#ba6b57"
+  - name: pig
+    fg: "#eba28f"
+objectives:
+  - teaser: Block in pig
+    goal:
+      - |
+        This porcine pest has been running amok.
+        Block the pig by a "monolith" on all four sides to capture it.
+      - Blocking on fewer than four sides will spook the pig, and he will escape!
+    condition: |
+      def isSurrounded = \n.
+          if (n > 0) {
+              blockedFwd <- blocked;
+              if blockedFwd {
+                  turn left;
+                  isSurrounded $ n - 1;
+              } {
+                  return false;
+              }
+          } {
+              return true;
+          }
+          end;
+
+      opponent <- robotnamed "opponent";
+      as opponent {
+        isSurrounded 4;
+      };
+  - teaser: Spooked
+    id: spooked
+    optional: true
+    hidden: true
+    goal:
+      - |
+        The pig was spooked by the capture attempt.
+    condition: |
+      def boolToInt = \b.
+          if b {1} {0}
+          end;
+
+      /**
+      Iterate 4 times (once for each direction)
+      */
+      def countBlocked = \n.
+          if (n > 0) {
+              isBlocked <- blocked;
+              // This assignment has to happen before the recursive
+              // "countBlocked" call due to #1032
+              let localBlockCount = boolToInt isBlocked in
+              turn left;
+              lastCount <- countBlocked $ n - 1;
+              return $ lastCount + localBlockCount;
+          } {
+              return 0;
+          }
+          end;
+
+      opponent <- robotnamed "opponent";
+      as opponent {
+          blockCount <- countBlocked 4;
+          return $ blockCount > 0 && blockCount < 4;
+      };
+robots:
+  - name: base
+    dir: [1, 0]
+    devices:
+      - 3D printer
+      - branch predictor
+      - calculator
+      - clock
```

Comment:
> I actually solved it by first arranging the monoliths one cell away from the pig, as in the model solution, and then building four robots to sit behind them and wait for a specific time using the time command, then all push simultaneously at the agreed-upon time. If you want to disallow that and force the use of watch then you could remove the clocks from the inventory.

I had overlooked the `time` command; that indeed makes it too easy.  However, the `wait` command, which is a necessary counterpart of the `watch` command, also requires the clock!  So I think I will want to refactor the capabilities required for `wait` vs. `time` as a prerequisite to landing this.

Reply:
#1261